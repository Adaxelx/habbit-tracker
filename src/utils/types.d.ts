export type GridProps = {
  events: Event[];
  month: number;
  year: number;
  moveDate: Function;
};

export interface Event {
  _id: string;
  daysOfWeek: number[];
  title: string;
  timeStart?: string;
  timeEnd?: string;
  dateStart: string;
  dateEnd: string;
  label?: string | Label;
  description?: string;
  userId: string;
  checked?: boolean;
}

export interface EventLoop extends Event {
  numericStart: number;
  numericEnd: number;
}

export type Label = {
  _id: string;
  title: string;
  color: string;
};

export type CalendarTile = {
  day: number;
  events: EventLoop[];
  id: number;
};

export enum SIDES {
  LEFT = -1,
  RIGHT = 1,
}
