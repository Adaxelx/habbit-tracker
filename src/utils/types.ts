export type GridProps = {
  events: Event[];
  month: number;
  year: number;
  moveDate: Function;
  handleDayChange: Function;
};

export interface Event {
  _id: string;
  daysOfWeek: number[];
  title: string;
  timeStart?: string;
  timeEnd?: string;
  dateStart: string;
  dateEnd: string;
  label?: string | Label;
  description?: string;
  userId: string;
  checked?: [
    {
      day: number;
      month: number;
      year: number;
      _id: string;
    },
  ];
}

export interface EventLoop extends Event {
  numericStart: number;
  numericEnd: number;
}

export type Label = {
  _id: string;
  title: string;
  color: string;
  userId: string;
};

export type CalendarTile = {
  day: number;
  events: EventLoop[];
  id: number;
};

export enum SIDES {
  LEFT = -1,
  RIGHT = 1,
}

export type Color = { color?: string; id?: string };

export type NavProps = {
  header: string;
  moveDate: Function;
};

export interface DayCardProps {
  day: string;
  token: string | undefined;
}
