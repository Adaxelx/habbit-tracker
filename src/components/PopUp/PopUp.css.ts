import styled from 'styled-components';

export const StyledWrapper = styled.div`
  width: ${({ theme }) => `calc(100% - ${theme.margin.l})`};
  height: ${({ theme }) => `calc(100vh - ${theme.margin.l})`};
  position: fixed;
  background-color: ${({ theme }) => theme.colors.modalBackground};
  z-index: ${({ theme }) => theme.zIndex.max};
  display: flex;
  flex-direction: column;
`;

export const StyledHeader = styled.h2`
  width: 100%;
  text-align: center;
  font-size: ${({ theme }) => theme.font.sizes.ml};
`;

export const StyledCloseButton = styled.button`
  width: 50px;
  height: 50px;
  position: relative;
  align-self: flex-end;
  border: none;
  background-color: transparent;
`;

interface Line {
  rotate: number;
}

export const StyledLine = styled.div<Line>`
  height: 100%;
  width: 2px;
  position: absolute;
  top: 0%;
  left: 50%;
  background-color: ${({ theme }) => theme.colors.error.main};
  transform: ${({ rotate }) => `rotate(${rotate}deg)`};
`;
