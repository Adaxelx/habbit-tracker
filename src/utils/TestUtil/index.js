export { default } from './TestUtil.ts';
