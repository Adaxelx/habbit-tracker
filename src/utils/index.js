export { default as TestUtil } from './TestUtil';
export { default as RenderWithRouter } from './RenderWithRouter';
export * from './validators';
export * from './types.d';
