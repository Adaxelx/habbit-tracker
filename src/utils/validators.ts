export const createRequiredObject = (name: string) => ({
  required: `Field ${name} is required.`,
});

export const validatePattern = (pattern: RegExp) => ({
  pattern: {
    value: pattern,
    message: 'Incorrect characters.',
  },
});

export const emailValidation = validatePattern(
  // eslint-disable-next-line no-control-regex
  /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/,
);

export const createRestrictedLengthObject = (name: string) => ({
  ...createRequiredObject(name),
  minLength: {
    value: 6,
    message: `Field ${name} must have at least 6 characters.`,
  },
  maxLength: {
    value: 30,
    message: `Field ${name} must have maximum 30 characters.`,
  },
});

const timeValidFuntion = (timeLess: string | undefined, time: string | undefined) => {
  const message = 'Your time is not in boundaries.';
  if (timeLess && time) {
    const [hours, minutes] = time.split(':');
    const [hoursMin, minutesMin] = timeLess.split(':');
    if (hoursMin < hours) {
      return true;
    }
    if (hoursMin === hours) {
      if (minutesMin < minutes) {
        return true;
      }
      return message;
    }
    return message;
  }
  return message;
};

export const createTimeValidation = (min: string | undefined, max: string | undefined) => ({
  validate: {
    min: (value: string) => timeValidFuntion(min, value),
    max: (value: string) => timeValidFuntion(value, max),
  },
});
