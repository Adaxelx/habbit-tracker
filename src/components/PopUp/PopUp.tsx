import React, { MouseEventHandler } from 'react';
import { StyledWrapper, StyledHeader, StyledCloseButton, StyledLine } from './PopUp.css';

interface PopUpProps {
  open: boolean;
  handleClose: MouseEventHandler<HTMLButtonElement>;
  header: string;
  children: React.ReactChild;
}

const PopUp = ({ open, handleClose, header, children }: PopUpProps) =>
  open ? (
    <StyledWrapper>
      <StyledCloseButton type="button" onClick={handleClose}>
        <StyledLine rotate={45} />
        <StyledLine rotate={-45} />
      </StyledCloseButton>
      <StyledHeader>{header}</StyledHeader>
      {children}
    </StyledWrapper>
  ) : null;

export default PopUp;
