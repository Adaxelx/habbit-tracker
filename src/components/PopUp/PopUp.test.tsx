import React from 'react';
import 'jest-styled-components';
import { TestUtil } from 'utils';
import PopUp from './PopUp';

describe('PopUp', () => {
  let util: TestUtil;
  let props;
  beforeEach(() => {
    jest.clearAllMocks();
    util = new TestUtil(<PopUp {...props} />);
  });

  it('should match snapshot', () => {
    expect(util.render.asFragment()).toMatchSnapshot();
  });
});
