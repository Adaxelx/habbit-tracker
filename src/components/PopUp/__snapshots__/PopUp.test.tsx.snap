// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopUp should match snapshot 1`] = `<DocumentFragment />`;
